### PR TITLE
Trozensztrauch/handover2

### PR DESCRIPTION
**Pull request by trozensztrauch**
We would like to support 'disable handover' functionality in NEAT core, which for SCTP connections means disabling multihoming feature and for MPTCP means silent fallback to TCP.

With this pull request we:
* use flow->isSCTPMultihoming field for both SCPT and MPTCP, thus we renamed it to something more general (flow->isMultihoming)
* introduce runtime check for MPTCP support (check for value in /proc/sys/net/mptcp/mptcp_enable); MPTCP can be globaly enabled/disabled or controlled by applications for each connection
* introduce silent fallback to TCP for MPTCP connections when multihoming is disabled
* MPTCP connection fails if MPTCP is globally disbled, TCP connection fails if MPTCP is globaly enabled
* introduce possibility to build NEAT with MPTCP support on machines where MPTCP is not installed
  (cmake -DMPTCP_SUPPORT flag)

### CODE REVIEW

File: neat_core.c [hunk 1]
```diff
@@ -5247,6 +5243,7 @@ neat_connect(struct neat_he_candidate *candidate, uv_poll_cb callback_fx)
     const char *algo;
 #endif
     int enable = 1;
+    int disable = 0;
```
Comment (naeemk) [inline neat_core.c]: @trozensztrauch `travis-ci` [fails](https://travis-ci.org/NEAT-project/neat/jobs/292060833) due to this unused variable, even though there is no git conflict. Could you please fix asap and PR?

Comment (trozensztrauch) [reply neat_core.c]: Thanks. I have just pushed a commit that should fix the problem. Let see if passes travis check now.

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
 # ignore build directory
+.vscode/
```
Comment (naeemk) [inline .gitignore]: @trozensztrauch: this looks unnecessary.

Comment (trozensztrauch) [reply .gitignore]: Great. I have removed that line.

**Pull request merged**